leads/onboarding.md

```diff
@@ -15,7 +15,7 @@ To ensure quality communication SIG Release Chairs and Technical Leads should:
   - kubernetes-release-team:
     https://groups.google.com/a/kubernetes.io/g/release-team
   - kubernetes-dev:
-    https://groups.google.com/forum/#!forum/kubernetes-dev
+    https://groups.google.com/a/kubernetes.io/group/dev
```

Comment:
maybe can apply the same here `https://groups.google.com/a/kubernetes.io/g/dev`

Reply:
Let me check again, I swear I did a find and replace all.

Reply:
Good catch @cpanato those where not markdown links